packages/core/contracts/optimistic-asserter/implementation/OptimisticAsserter.sol

```diff
@@ -153,20 +153,16 @@ contract OptimisticAsserter is OptimisticAsserterInterface, Lockable, Ownable {
     }
 
     // TODO think about the naming of this function and readAssertion
-    function getAssertion(bytes32 assertionId) public view returns (bool) {
-        Assertion memory assertion = assertions[assertionId];
-        // Return early if not using answer from resolved dispute.
-        if (assertion.disputer != address(0) && !assertion.ssSettings.useDisputeResolution) return false;
-        require(assertion.settled, "Assertion not settled"); // Revert if assertion not settled.
-        return assertion.settlementResolution;
+    function getAssertion(bytes32 assertionId) external view nonReentrantView() returns (bool) {
+        return _getAssertion(assertionId);
     }
 
-    function settleAndGetAssertion(bytes32 assertionId) public returns (bool) {
-        if (!assertions[assertionId].settled) settleAssertion(assertionId);
-        return getAssertion(assertionId);
+    function settleAndGetAssertion(bytes32 assertionId) external nonReentrant() returns (bool) {
```

Comment:
Separating internal methods follows the same pattern as in Optimistic Oracle.

Another option would be to skip re-entrancy guard in `settleAndGetAssertion` and not use internal `_settleAssertion` and `_getAssertion` methods as `settleAssertion` and `getAssertion` each has re-entrancy protection. This would be better for contract readability.

Reply:
Removed rentrancy guard here as it is done in `settleAssertion`.